executor/load_data.go

```diff
@@ -98,9 +95,84 @@ func (e *LoadDataExec) Open(ctx context.Context) error {
 	if e.loadDataInfo.insertColumns != nil {
 		e.loadDataInfo.initEvalBuffer()
 	}
+	dataStream := e.ctx.Value(LoadDataInput)
+	if dataStream == nil {
+		dataStream = ctx.Value(LoadDataInput)
+		if dataStream == nil {
+			return errors.New("ctx has no LoadDataInput data stream")
+		}
+		e.ctx.SetValue(LoadDataInput, dataStream)
+	}
 	return nil
 }
 
+// DoCmd do the load data work, input DataStream could
+// be clientConn or read file on server(non local mode)
+func (e *LoadDataExec) DoCmd(ctx context.Context) error {
+	var err error
+	loadDataInfo := e.loadDataInfo
+	// LoadDataInput default is server.clientConn,
+	// if non local mode is supported in the future,
+	// LoadDataInput value in Ctx should be reset in buildLoadData
+	input, ok := e.ctx.Value(LoadDataInput).(DataStream)
+	if !ok {
+		return errors.New("load data input is nil")
+	}
+	err = input.LoadPreCheck()
+	if err != nil {
+		return err
+	}
+	err = input.ReqData(loadDataInfo.Path)
+	if err != nil {
+		return err
+	}
+
+	var shouldBreak bool
+	var prevData, curData []byte
+	err = loadDataInfo.Ctx.NewTxn(ctx)
+	if err != nil {
+		return err
+	}
+	for {
+		curData, err = input.ReadFromStream()
+		if err != nil {
+			if terror.ErrorNotEqual(err, io.EOF) {
+				logutil.Logger(ctx).Error("input failed", zap.Error(err))
+				break
+			}
+		}
+		if len(curData) == 0 {
+			shouldBreak = true
+			if len(prevData) == 0 {
+				break
+			}
+		}
+		prevData, err = loadDataInfo.insertDataWithCommit(ctx, prevData, curData)
+		if err != nil {
+			break
+		}
+		if shouldBreak {
+			break
+		}
+	}
+	loadDataInfo.SetMessage()
+
+	if err != nil {
+		loadDataInfo.Ctx.StmtRollback()
+	} else {
+		err = loadDataInfo.CheckAndInsertOneBatch(ctx)
```

Comment:
This seems redundant in `insertDataWithCommit`, maybe we can extract some same code pattern.

Reply:
there is some little refactoration in #11533